core/job/service/job_service.go

```diff
@@ -524,33 +563,58 @@ func (j *JobService) bulkDelete(ctx context.Context, jobTenant tenant.Tenant, to
 	j.logger.Debug("deleting %d jobs of project [%s] namespace [%s]", len(toDelete), jobTenant.ProjectName(), jobTenant.NamespaceName())
 	me := errors.NewMultiError("bulk delete specs errors")
 	deletedJob := 0
+	toDeleteMap := map[job.FullName]bool{}
```

Comment:
since this is being used in both places, let's extract the map creation and make it a `job.Specs` behavior?